lib/src/builder/generation/impl_generation.dart

```diff
@@ -106,18 +105,34 @@ class ImplGenerator {
         );
       }
 
-      outputContentsBuffer
-        ..writeln('// React component factory implementation.')
-        ..writeln('//')
-        ..writeln('// Registers component implementation and links type meta to builder factory.')
-        ..writeln('final $generatedComponentFactoryName = registerComponent(() => new $componentClassImplMixinName(),')
-        ..writeln('    builderFactory: $factoryName,')
-        ..writeln('    componentClass: $componentClassName,')
-        ..writeln('    isWrapper: ${componentDeclNode.meta.isWrapper},')
-        ..writeln('    parentType: $parentTypeParam,$parentTypeParamComment')
-        ..writeln('    displayName: ${stringLiteral(factoryName)}')
-        ..writeln(');')
-        ..writeln();
+      if (isComponent2) {
+        outputContentsBuffer
+          ..writeln('// React component factory implementation.')
+          ..writeln('//')
+          ..writeln('// Registers component implementation and links type meta to builder factory.')
+          ..writeln('final $generatedComponentFactoryName = registerComponent2(() => new $componentClassImplMixinName(),')
+          ..writeln('    builderFactory: $factoryName,')
+          ..writeln('    componentClass: $componentClassName,')
+          ..writeln('    isWrapper: ${componentDeclNode.meta.isWrapper},')
+          ..writeln('    parentType: $parentTypeParam,$parentTypeParamComment')
+          ..writeln('    displayName: ${stringLiteral(factoryName)},')
+          ..write(declarations.component2.meta.isErrorBoundary == true ? '    skipMethods: [],\n' : '')
```

Comment:
I think this would be a little easier to grok if it was in an if block instead of a ternary, and I think that a code comment is justified since `skipMethods` isn't exactly a "slam dunk" mental model mapping directly to something that seems necessary to allow error boundary lifecycle methods to work.

```dart
if (isComponent2) {
  outputContentsBuffer
    ..writeln('// React component factory implementation.')
    ..writeln('//')
    ..writeln('// Registers component implementation and links type meta to builder factory.')
    ..writeln('final $generatedComponentFactoryName = registerComponent2(() => new $componentClassImplMixinName(),')
    ..writeln('    builderFactory: $factoryName,')
    ..writeln('    componentClass: $componentClassName,')
    ..writeln('    isWrapper: ${componentDeclNode.meta.isWrapper},')
    ..writeln('    parentType: $parentTypeParam,$parentTypeParamComment')
    ..writeln('    displayName: ${stringLiteral(factoryName)},');

  if (declarations.component2.meta.isErrorBoundary) {
    // Override `skipMethods` as an empty list so that
    // the `componentDidCatch` and `getDerivedStateFromError`
    // lifecycle methods are included in the component's JS bindings. 
    outputContentsBuffer
      ..writeln('    skipMethods: const [],');
  }

  outputContentsBuffer
    ..writeln(');')
    ..writeln();
}
```


> Also, note the use of `writeln` instead of `write` with a trailing `\n`, and the best practice of making empty list arguments to a function / method `const`.